toqito/channel_props/is_extremal.py

```diff
@@ -0,0 +1,64 @@
+"""Determines whether a quantum channel is extremal."""
+
+import numpy as np
+from numpy.linalg import matrix_rank
+
+from toqito.channel_ops.choi_to_kraus import choi_to_kraus
+
+
+def is_extremal(phi: np.ndarray | list[np.ndarray | list[np.ndarray]], tol: float = 1e-9) -> bool:
+    """Determine whether a quantum channel is extremal.
+
+    A channel with Kraus operators {A_i} is extremal if and only if
+    the set {A_i^† A_j : i,j=1,...,r} is linearly independent.
+
+    The channel can be provided as either:
+      - A Choi matrix (np.ndarray), which is converted to a flat list of Kraus operators.
```

Comment:
Since we provide the types in the argument, we don't need the types here in the docstring. This should be for a description about what these inputs pertain to for the purposes of the problem.